src/uPlot.js

```diff
@@ -1295,11 +1295,14 @@ export default function uPlot(opts, data, then) {
 
 			ctx.font         = axis.font[0];
 			ctx.fillStyle    = axis.stroke || hexBlack;									// rgba?
-			ctx.textAlign    = angle > 0 ? LEFT :
+			ctx.textAlign    = axis.align == 1 ? LEFT :
+					           axis.align == 2 ? RIGHT :
+					           angle > 0 ? LEFT :
 			                   angle < 0 ? RIGHT :
-			                   ori == 0 ? "center" : side == 3 ? RIGHT : LEFT;
+							   ori == 0 ? "center" :
```

Comment:
please leave the unrelated re-formatting here and below as it was. it was aligned it on purpose (pardon the pun?).